### DISCUSSION TITLE
onPaid listener not firing in popup.js

### DISCUSSION THREAD

**Discussion by FlashInThePan** [Glench/ExtPay#114] [General]
I have put the following in my manifest.js

{
    "content_scripts": [
        {
            "matches": ["https://extensionpay.com/*"],
            "js": ["ExtPay.js"],
            "run_at": "document_start"
        }
    ]
}

and the following in my popup.js:

extpay.onPaid.addListener(user => {
    console.log('user paid!')
})
(I noticed that this block is commented out in the V3 sample extension).

Unfortunately, the event listener isn't firing when subscription paid.

Any advice appreciated.

**Comment by Glench** [comment]
I would recommend putting onPaid listeners in the background file, not the popup because the popup often closes when the payment popup is opened.
